packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -216,13 +216,16 @@ export function authorizationServiceBuilder(
         logger,
         authData,
       }: WithLogger<AppContext<UIAuthData | M2MAuthData | M2MAdminAuthData>>
-    ): Promise<{ client: Client; showUsers: boolean }> {
+    ): Promise<WithMetadata<{ client: Client; showUsers: boolean }>> {
       logger.info(`Retrieving Client ${clientId}`);
       const client = await retrieveClient(clientId, readModelService);
       assertOrganizationIsClientConsumer(authData, client.data);
```

Comment:
```suggestion
      const { data: client, metadata } = await retrieveClient(clientId, readModelService);
      assertOrganizationIsClientConsumer(authData, client);
```

Minor, but I think it makes this (and the return below) a bit more readable